lib/handler/mruby.c

```diff
@@ -849,28 +864,20 @@ void h2o_mruby_run_fiber(h2o_mruby_context_t *ctx, mrb_value receiver, mrb_value
     if (generator == NULL)
         goto Exit;
     assert(generator->req != NULL);
+
     if (generator->req->_generator != NULL) {
         mrb->exc = mrb_obj_ptr(mrb_exc_new_str_lit(mrb, E_RUNTIME_ERROR, "unexpectedly received a rack response"));
         goto GotException;
     }
+
     send_response(generator, status, output, is_delegate);
+    if (mrb->exc != NULL)
```

Comment:
How about returning a int from `send_response` that indicates if the function has succeeded (zero) or failed (non-zero)?

I suggest this is for consistency. For example, `h2o_mruby_iterate_headers` is designed as such. If the function returns a failure, then we consult `mrb->exc`.

Honestly speaking, I do not have a strong preference on if we should `mrb->exc` being set to non-NULL as as indication of error, or if we should use a return value. However, I do prefer having consistency.

Reply:
fixed at https://github.com/h2o/h2o/pull/1493/commits/60101c0f558da1b9be94f3285252bd32421861df